### ISSUE TITLE
Support settings sections with | operator

### ISSUE THREAD

**Issue by captain0xff** [sublimelsp/LSP#2560]
The roslyn lsp server (https://github.com/dotnet/roslyn) uses the `|` symbol to separate settings levels (like `csharp|completion.dotnet_show_completion_items_from_unimported_namespaces`). The sublime lsp client doesn't seem to support this. 
It will be nice if we can have a customizable approach here so that new operators can be easily added.

**Comment by rwols** [comment]
I think this should be doable with ST's [score_selector](https://www.sublimetext.com/docs/api_reference.html#sublime.score_selector).

**Comment by jwortmann** [comment]
Could you explain what in particular is not supported?

I don't think there is anything needed to do here. This syntax seems to be used for the section names in `workspace/configuration`. We are able to split on dots if the requested `section` is a sub-path https://github.com/sublimelsp/LSP/blob/9be040dfc9d6bad3a7d4bb7c41f94f74ac4f0003/plugin/core/collections.py#L45
but in general the section names can be arbitrary strings.

So you should be able to just put those setting names into your config:

```jsonc
{
  "clients": {
    "roslyn": {
      // ...
      "settings": {
        "csharp|completion.dotnet_show_completion_items_from_unimported_namespaces": true,
        "csharp|symbol_search.dotnet_search_reference_assemblies": "somevalue",
        // ...
      }
    }
  }
}
```

**Comment by captain0xff** [comment]
Yep, you are right. I also found the same after looking at the VSCode extension. Actually, I am pulling some hacks in the server which deformed the message. While sharing the log I disabled the hack which also fixed this.

**Comment by captain0xff** [comment]
Sorry, for the false report.
